compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1425,59 +1425,26 @@ fn check_where_clauses<'tcx>(wfcx: &WfCheckingCtxt<'_, 'tcx>, span: Span, def_id
     let tcx = wfcx.tcx();
 
     let predicates = tcx.predicates_of(def_id.to_def_id());
-    let generics = tcx.generics_of(def_id);
-
-    let is_our_default = |def: &ty::GenericParamDef| match def.kind {
-        GenericParamDefKind::Type { has_default, .. }
-        | GenericParamDefKind::Const { has_default, .. } => {
-            has_default && def.index >= generics.parent_count as u32
-        }
-        GenericParamDefKind::Lifetime => {
-            span_bug!(tcx.def_span(def.def_id), "lifetime params can have no default")
-        }
-    };
 
     // Check that concrete defaults are well-formed. See test `type-check-defaults.rs`.
     // For example, this forbids the declaration:
     //
     //     struct Foo<T = Vec<[u32]>> { .. }
     //
     // Here, the default `Vec<[u32]>` is not WF because `[u32]: Sized` does not hold.
-    for param in &generics.own_params {
-        match param.kind {
-            GenericParamDefKind::Type { .. } => {
-                if is_our_default(param) {
-                    let ty = tcx.type_of(param.def_id).instantiate_identity();
-                    // Ignore dependent defaults -- that is, where the default of one type
-                    // parameter includes another (e.g., `<T, U = T>`). In those cases, we can't
-                    // be sure if it will error or not as user might always specify the other.
-                    if !ty.has_param() {
-                        wfcx.register_wf_obligation(
-                            tcx.def_span(param.def_id),
-                            Some(WellFormedLoc::Ty(param.def_id.expect_local())),
-                            ty.into(),
-                        );
-                    }
-                }
-            }
-            GenericParamDefKind::Const { .. } => {
-                if is_our_default(param) {
-                    // FIXME(const_generics_defaults): This
-                    // is incorrect when dealing with unused args, for example
-                    // for `struct Foo<const N: usize, const M: usize = { 1 - 2 }>`
-                    // we should eagerly error.
```

Comment:
I've removed this FIXME as the provided snippet now does eagerly error (with `evaluation of constant value failed`). Note that I couldn't find any UI (regression) test for this, so I guess I should add one? People more familar with const generics & eval, please let me know if there's more context to this.

Reply:
Under `feature(generic_const_exprs)` this would be some anon const represented as `ConstKind::Unevaluated(def, [N, M])` which would be considered to have generic parameters. e.g. [this example](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=81a27944eb6e16516fc3309a88ae359e)
```rust
#![feature(generic_const_exprs)]

struct Foo<T, const N: usize = { 0_usize - 1 }>(T);
```

Reply:
Oh, you're absolute right! I forgot about GCE for a sec. Should I re-add the FIXME under GCE instead? Or does that not matter? I'm never quite sure if you're serious with removing GCE from the compiler once mGCE has settled ^^

Reply:
Re-added the FIXME under GCE (was CGD).